util/perl/OpenSSL/copyright.pm

```diff
@@ -23,11 +23,11 @@ sub year_of {
 
     # See if git's available
     open my $FH,
-       "git log -1 --date=format:%Y --format=format:%ad $file 2>/dev/null|"
+       "git log -1 --date=short --format=format:%cd $file 2>/dev/null|"
            or return $YEAR;
     my $LINE = <$FH>;
     close $FH;
-    chomp($LINE);
+    $LINE =~ s/^([0-9]*)-.*/$1/;
```

Comment:
@bernd-edlinger can you please explain the change? I.E. do you think it will help the next time?

Reply:
At least some get versions don't understand the `--date=format:%Y` and frankly I don't find where this
form of the `--date` option is documented at all.
OTOH `--date=short` should always work. we just have to parse the YYYY-MM-DD format and extract the year.
When this command is not understood we have a different result which makes the make update change to 2022.
Additionally I think this is better to use the commit date instead of the author date, since the merge may be
in a later year than the original commit.

Reply:
The update of the .pl files is only necessary because of #17398, otherwise that commit would need to be reverted.

Reply:
> The update of the .pl files is only necessary because of #17398, otherwise that commit would need to be reverted.

That's actually a question - should it be reverted?

Reply:
Maybe, yes.  I don't have a strong opinion.

Reply:
> Additionally I think this is better to use the commit date instead of the author date, since the merge may be
in a later year than the original commit.

From a copyright perspective, the date when the code was written looks more interesting, not when it was merged. But I'm happy with either.

Reply:
Suppose `PR 1` contains a file `f.pl` committed in 2021, and `PR 2` also contains `f.pl` committed in 2022. `PR 2` is then merged followed by `PR 1`. The copyright will show as 2021 instead of the legally desirable 2021, 2022 or 2021-2022, or bare minimum 2022. The copyright statement is (arguably) more relevant at the whole product level rather than for an individual file based on when the copyright content is registered. I do not think the commit date is effective in this change.

Reply:
Yes, but if I remember correctly, some of my PRs took years for a merge, and as soon as they got
finally merged, the CIs failed immediatley, until another fresh commit is merged.
And often the author date, is when the first commit was cretated, but neiter `--fixup` commits nor
re-base done by the author touch the author date any more.

Reply:
So a commit that passed CI in 2021, but gets merged in 2022 will still create problems?

Reply:
> Suppose PR 1 contains a file f.pl committed in 2021, and PR 2 also contains f.pl committed in 2022. PR 2 is then merged followed by PR 1. The copyright will show as 2021 instead of the legally desirable 2021, 2022 or 2021-2022, or bare minimum 2022

No, if the f.pl was **merged** in 2022, that will be the commit date, so the generated files will be (C) 2022.

Reply:
> So a commit that passed CI in 2021, but gets merged in 2022 will still create problems?

Yes, but only if that touches a source file, at which time the copyright year in the source file should
manually be updated, that will usually be the current year, of course.  When that is done
the CI fill flag the generated file's copyright statement.
I for instance look always in the files that are touched when I review a PR, and request changing
the copyright year of that specific file.

Reply:
This looks like something that should really be done by a git clean/smudge filter rather than the CI; although the impact of doing that to contributors is a bit high (installing the filters).

Reply:
So my understanding is that no matter what we do, there will always be a way to break it. It's up to the one merging to make sure that after merging things still work properly.

Reply:
> So my understanding is that no matter what we do, there will always be a way to break it. It's up to the one merging to make sure that after merging things still work properly.

Pretty much. On my main project, we manually commit copyright management information because of this issue.

Reply:
No, currently the situation is really a mess.
1.1.1 uses a completely different approach, in the CIs it is just the wall clock time,
while master uses a git command that may or may not be understood.

Reply:
Another approach is to use an annotated git tag with the copyright notice in the annotation on the release commit and use git describe to find the tag and process it via CI. At least that is predictable.

Reply:
If you guys agree, I will update the PR to not touch the `.pl` files, and restore the 2021 copyright in the generated files
instead.

Reply:
Hmm, the CI failure of  96b5b11 is a bit unexpected.
Apparently some commit dates are changed when the CI re-bases the PR,
but I am not able to reproduce that at home...

Reply:
Hmm, the CI uses `git fetch --depth=1` this fetches no history beyond the merge commit, which is always new.

Reply:
> Hmm, the CI uses `git fetch --depth=1` this fetches no history beyond the merge commit, which is always new.

That is pretty common. In Jenkins, I force all history and tags mostly to get the tags. Surely there must be a way to change this.

Reply:
Right, adding `with: fetch-depth:0` in the right place fixes it.